test/benchmarks/benchmark_compare258.cc

```diff
@@ -0,0 +1,73 @@
+/* benchmark_compare258.cc -- benchmark compare258 variants
+ * Copyright (C) 2020 Nathan Moinvaziri
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include <benchmark/benchmark.h>
+
+#include "zbuild.h"
+#include "zutil.h"
+
+#include "cpu_features.h"
+
+typedef uint32_t (*compare258_func)(const unsigned char *src0, const unsigned char *src1);
+
+static inline void compare258_bench(benchmark::State& state, compare258_func compare258) {
+    unsigned char str1[274];
+    unsigned char str2[274];
+    uint32_t len;
+    int32_t x = 0;
+
+    memset(str1, 'a', sizeof(str1));
+    memset(str2, 'a', sizeof(str2));
+
+    while (state.KeepRunning()) {
+        str2[x] = 0;
```

Comment:
This shares the previous issue of faster routines rolling over the the len counter more times, landing at unpredictable states waiting for the previous iteration to complete.  We'll need to come up with a way to instead parameterize the length of this byte string comparison so that it's fixed within the infinite timing loop.

Reply:
So we should use the same method of increasing by power of 2 up to 256..?

Reply:
That's sensible for input lengths.  Since the timing will be data dependent we'll need a way to modify where the first unmatched character occurs prior to entering the benchmark loop.

Reply:
uh duh, you're basically doing that with memsets.  So it's just a matter of modifying where the memset stops are based on range.